### ISSUE TITLE
Use 'is' instead of 'Equals(StringComparison.Ordinal)' for string comparisons

### ISSUE THREAD

**Issue by RalfKoban** [RalfKoban/MiKo-Analyzers#1493]
To ease reading, when comparing to an ordinal string, we should report an issue if `.Equals(StringComparison.Ordinal)` is used instead of the `==` equality operator.

Following code should trigger issues:
```c#
string s1;
string? s2;

var a = s1.Equals("someValue", StringComparison.Ordinal);
var b = s2.Equals("someValue", StringComparison.Ordinal);
```

It should be converted into:
```c#
string s1;
string? s2;

var a = s1 == "someValue";
var b = s2 == "someValue";
```
or alternatively (preferred):
```c#
string s1;
string? s2;

var a = s1 is "someValue";
var b = s2 is "someValue";
```
